hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -86,6 +88,7 @@
 
   protected HoodieTimer timer;
   protected WriteStatus writeStatus;
+  protected HoodieRecordLocation newRecordLocation;
```

Comment:
Note to reviewer: record location for a given handle applies to all records. so just reusing the same instance rather than creating new instance for every record.

Reply:
I think it is also necessary to obtain the location from every record. The index can not only reach the file level, but also the rowGroup level and even the row level

Reply:
I got it wrong. I thought that a WriteStatus only has one location instance. Now it seems that there is no great change from obtaining the loaction from HoodieRecord. But in some partition change scenarios, we may need to use HoodieRecord's operation to determine the curd of the index